pkg/overlay/server_test.go

```diff
@@ -71,3 +75,171 @@ func TestServer(t *testing.T) {
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     1,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+			totalNodes:            2,
+		},
+		{
+			name:                  "case: 50% new nodes requested",
+			requestedNodeAmt:      2,
+			reputableNodes:        2,
+			newNodePercentage:     0.5,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+			totalNodes:            3,
+		},
+		{
+			name:                  "case: more than required reputable nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			newNodePercentage:     0,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+			totalNodes:            4,
+		},
+		{
+			name:                  "case: fewer than required reputable nodes",
+			requestedNodeAmt:      4,
+			newNodePercentage:     0,
+			reputableNodes:        3,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+			totalNodes:            4,
+		},
+		{
+			name:                  "case: zero reputable nodes found, only new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        0,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+			newNodePercentage:     1,
+			totalNodes:            3,
+		},
+		{
+			name:                  "case: fewer than required new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			newNodePercentage:     0.5,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+			totalNodes:            4,
+		},
+		{
+			name:                  "case: more than required new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        2,
+			newNodePercentage:     0.5,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+			totalNodes:            5,
+		},
+		{
+			name:                  "case: low percentage of new nodes",
+			requestedNodeAmt:      3,
+			reputableNodes:        1,
+			newNodePercentage:     0.01,
+			expectedResultLength:  1,
+			newNodeAuditThreshold: 1,
+			totalNodes:            4,
+		},
+		{
+			name:                  "case: high percentage of new nodes",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     3,
+			expectedResultLength:  4,
+			newNodeAuditThreshold: 1,
+			totalNodes:            5,
+		},
+		{
+			name:                  "case: zero new nodes found, only reputable nodes",
+			requestedNodeAmt:      5,
+			reputableNodes:        10,
+			newNodePercentage:     0,
+			expectedResultLength:  5,
+			newNodeAuditThreshold: 1,
+			totalNodes:            10,
+		},
+	} {
+
+		planet, err := testplanet.New(t, 1, tt.totalNodes, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		planet.Start(ctx)
+		// we wait a second for all the nodes to complete bootstrapping off the satellite
+		time.Sleep(2 * time.Second)
+
+		satellite := planet.Satellites[0]
+
+		server := overlay.NewServer(satellite.Log.Named("overlay"), satellite.Overlay.Service,
```

Comment:
If we make server have the defaults and service the the configuration we don't need to recreate testplanet for every test.